python/ray/experimental/gpu_object_manager/gpu_object_manager.py

```diff
@@ -186,33 +190,27 @@ def fetch_object(self, obj_id: str, use_object_store: bool = True):
         tensor_transport_manager = get_tensor_transport_manager(
             tensor_transport_backend
         )
-        if use_object_store or not tensor_transport_manager.is_one_sided():
+        tensor_transport_meta = gpu_object_meta.tensor_transport_meta
+        use_object_store = (
+            tensor_transport == TensorTransportEnum.OBJECT_STORE
+            or isinstance(tensor_transport_meta, ObjectRef)
```

Comment:
Hmm I think instead of implicitly choosing to use object store for non-NIXL backends, we should just error if a collective-based transport is going to be used and ask the user to specify "object_store" instead. The system behavior is more predictable if we can keep the default transport consistent between different tensor transports.